hil/ext/switches/brocade.py

```diff
@@ -265,6 +271,89 @@ def _disable_native_tag(self, interface):
+        body = '<bna-config-cmd><src>running-config</src><dest>startup-config \
+                </dest></bna-config-cmd>'
+        response = self._make_request('POST', url, data=body)
+
+        # check if it succesfully completed and log it
+        tag = '{urn:brocade.com:mgmt:brocade-ras}session-id'
+        root = etree.fromstring(response.text)
+        session_id = root.find(tag).text
+        if self._check_bna_config_status(session_id):
+            logger.debug('Saved running config to startup config')
+        else:
+            logger.debug('Configuration not saved in reasonable time')
+
+    def get_config(self, config_type):
+        # This is the only method that uses pexpect to connect to the switch
+        # because the REST API does not return the startup config file.
+        # This method is only used by the test suite, the rest of the switch
+        # functionality does not care about this method.
+
+        # this is how we get the running config file using the API
+        # I couldn't find a way to get the startup config file similarly.
+        # url = self.hostname + '/rest/config/' + config_type
+        # response = self._make_request('GET', url)
+
+        from hil.ext.switches._console import login, get_prompts
+
+        # is this safe to assume that there will be http in front of the
+        # hostname? also create a deep copy rather than modifying the original
+        # object
+        original_hostname = self.hostname
+        self.hostname = self.hostname.replace('http://', '')
+        console = login(self)
+        self.hostname = original_hostname
+
+        console.sendline('')
+        prompts = get_prompts(console)
+
+        console.sendline('terminal length 0')
+        console.expect(r'[\r\n]+.+# ')
+
+        # dunno why, but without asking twice, I didn't get the desired
+        # response. I'll debug this
+        console.sendline('show ' + config_type + '-config')
+        console.expect(r'[\r\n]+.+# ')
+        console.sendline('show ' + config_type + '-config')
+        console.expect(r'[\r\n]+.+# ')
+
+        config = console.after
+        console.sendline('terminal length 40')
+        console.sendline('exit')
+        return config
+
+    def _check_bna_config_status(self, id):
+        """ This function checks if the bna_config_command was successful by
+        polling on the session_id returned by running the command
+
+        Args:
+            id: the id of configuration command whose status we want to check
+
+        Returns a boolean indicating success/failure.
+        """
+
+        url = self.hostname + '/rest/' +  \
+            'operational-state/bna-config-cmd-status'
+        body = '<bna-config-cmd-status><session-id>%s</session-id> \
+                </bna-config-cmd-status>' % id
+        tag = '{urn:brocade.com:mgmt:brocade-ras}status'
+
+        # Check status 6 times and sleep for 2 seconds before checking again.
```

Comment:
this means, that every `modify_port` and `revert_port` operation will be blocked for up to 12 seconds. Should I be calling this method with a new thread, @zenhack ?

Reply:
How long does it typically take? I think the answer depends.

Reply:
It takes 8 seconds typically. To avoid complicated code, I could just not check this at all.